Decrease waiting-for-state-timeout to 2s

chbatey opened the pull request:
The timeout can be likely to happen for small clusters when the read
majority needs to go to every node

Refs #26710

chbatey committed:
Decrease waiting-for-state-timeout to 2s

The timeout can be likely to happen for small clusters when the read
majority needs to go to every node

Refs #26710

patriknw reviewed (approved):
LGTM

patriknw commented on akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md:
```diff
@@ -200,6 +200,12 @@ To disable passivation you can use configuration:
+
+This has been reduced to speed up ShardCoordinator initialization in smaller clusters.
+The read from ddata is a ReadMajority, for small clusters (< majority-min-cap) every node needs to respond
+so is more likely to timeout if there are nodes restarting i.e. when there is a rolling re-deploy happening.
```
Maybe we should have a separate section in migration guide about configuration (and behavior) changes that doesn't require any "migration" changes from the user?

On the other hand it's also nice to group things per module (Sharding, Streams).

helena replied:
Useful to add to the migration guide and note in the respective doc sections.

chbatey replied:
I think a separate section for configuration changes makes sense

chbatey replied:
I restructured the migration guide to put all changes of behavior that don't require user action into their own section so we have fewer top-level headings

helena commented on akka-cluster-sharding/src/main/resources/reference.conf:
```diff
@@ -108,9 +108,12 @@ akka.cluster.sharding {
 
-  # Timeout of waiting the initial distributed state (an initial state will be queried again if the timeout happened)
+  # Timeout of waiting the initial distributed state for the shard coordinator (an initial state will be queried again if the timeout happened)
+  # and for a shard to get its state when remembered entities is enabled
```
This is run-on-sentency, doesn't quite read as well as it could.

helena commented on akka-cluster-sharding/src/main/resources/reference.conf:
```diff
@@ -108,9 +108,12 @@ akka.cluster.sharding {
+  # Timeout of waiting the initial distributed state for the shard coordinator (an initial state will be queried again if the timeout happened)
+  # and for a shard to get its state when remembered entities is enabled
+  # The read from ddata is a ReadMajority, for small clusters (< majority-min-cap) every node needs to respond
+  # so is more likely to time out if there are nodes restarting i.e. when there is a rolling re-deploy happening
```
```suggestion
  # so is more likely to time out if there are nodes restarting i.e. during a rolling re-deploy.
```

helena commented on akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md:
```diff
@@ -200,6 +200,12 @@ To disable passivation you can use configuration:
+
+This has been reduced to speed up ShardCoordinator initialization in smaller clusters.
+The read from ddata is a ReadMajority, for small clusters (< majority-min-cap) every node needs to respond
+so is more likely to timeout if there are nodes restarting i.e. when there is a rolling re-deploy happening.
```
Same suggested change as above, including i.e. -> e.g.

chbatey committed:
Restructure migration guide for 2.6

Separate out changes that require no user action unless the old
behavior is preferred

chbatey commented:
PLS BUILD

chbatey committed:
Merge branch 'master' into issue-26710

chbatey committed:
Merge branch 'master' into issue-26710

raboof commented on akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md:
```diff
@@ -200,11 +175,11 @@ Classic remoting is deprecated but can be used in `2.6.` Any configuration under
-A full cluster restart is required to change to Artery.
+## Configuration and behavior changes
+
+The following documents configuration changes and behavior changes where no action is required. In come cases the old
```
```suggestion
The following documents configuration changes and behavior changes where no action is required. In some cases the old
```

chbatey replied:
Thanks!

chbatey committed:
Update akka-docs/src/main/paradox/project/migration-guide-2.5.x-2.6.x.md

Co-Authored-By: Arnout Engelen <github@bzzt.net>
